account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -18,19 +19,19 @@ class AccountInvoice(models.Model):
     )
     web_url = fields.Char()
 
-    def _get_mail_template_id(self):
-        try:
-            res = self.env.ref(
-                "account_invoice_validate_send_email.email_template_customer_invoice_validated"
-            ).id
-        except ValueError:
-            res = False
-        return res
+    def _get_mail_template(self):
+        self.ensure_one()
+        if self.company_id.invoice_mail_template_id:
+            return self.company_id.invoice_mail_template_id
+        else:
+            return False
 
     @api.multi
     def get_mail_compose_message(self):
         self.ensure_one()
-        template_id = self._get_mail_template_id()
+        template = self._get_mail_template()
+        if not template:
+            raise UserError(_("company_id cannot detect"))
```

Comment:
@tack-yabeQuartile Should we raise an error here given that the process (invoice validation) is expected to be triggered by the automatic workflow most of the time?  In other words, would raising error here be of any use?  I was thinking maybe just `return {}` would do instead, in which case the invoice validation should still complete without sending out an email when there is no template found. I think that is actually fine.  What do you think?

Reply:
@yostashiro Thank you for comments. After thinking back about auto workflow, I agree that this error might not be observed by user companies. So I'll delete this if-block and related operations.
I'd like to notify in README that an email doesn't sent when here is no template found.

Reply:
@tack-yabeQuartile We should still keep the if block or there will be a server error when no template is found. So, just return an empty dictionary if template is not available.